packages/rxfire/firestore/collection/index.ts

```diff
@@ -80,22 +80,22 @@ function processIndividualChange(
     case 'modified':
       if (
         combined[change.oldIndex] == null ||
-        combined[change.oldIndex].doc.id === change.doc.id
+        combined[change.oldIndex].doc.ref.isEqual(change.doc.ref)
       ) {
         // When an item changes position we first remove it
         // and then add it's new position
         if (change.oldIndex !== change.newIndex) {
           combined.splice(change.oldIndex, 1);
           combined.splice(change.newIndex, 0, change);
         } else {
-          combined[change.newIndex] = change;
+          combined.splice(change.newIndex, 1, change);
```

Comment:
Splicing doesn't create a new array, but modify the existing array. Do you mean to use slice() ?

Reply:
Derp, yes, good catch. Yes I meant to use slice... further it should be used in the other spots too. Will make that change, thanks @Feiyang1

Reply:
I'll write some tests for this while I'm at it.